Topic/evenbetterwebsite

aj-michael opened the pull request:
I am progressively adding more features to the site. User registration and log in now works. The dashboard page requires that you log in first, or else you will be redirected.

aj-michael committed:
Changed code style.

It looks like I'll be taking over development of this server for a bit, so I got rid of the pesky code formatting warnings in my IDE.

aj-michael committed:
Removed old images.

aj-michael committed:
Removed package name.

The scala templating engine does not play nice with package names.

aj-michael committed:
Added Play-style authentication.

I will be progressively moving the UI from /public/website/pages to /app/views.

aj-michael commented:
David, review this whenever over the next few days. I will keep adding to it until you merge it, so the longer you wait, the more changes there will be.

robinsdm commented on app/controllers/ApplicationController.java:
```diff
@@ -0,0 +1,113 @@
+    @RequireJson() public Promise<Result> createApplication() {
+        JsonNode jsonNode = request().body().asJson();
+        if (!jsonNode.isObject()) {
+            return Promise.pure(badRequest("Json sent was not a Json objects"));
```
By the way, when Carter was around he brought up a concern with these badRequest functions. The problem was that they return strings and not JSON to the user. To keep consistency with pathfinder-server they should probably be changed to return a JSON encoded string.

aj-michael replied:
I've switched away from using JSON. I'm using Play's built in tempting system instead of making an Angular app, so no custom ajax calls to the server are needed.

aj-michael replied:
And I don't really see a reason to need to be consistent with pathfinder-server.

aj-michael committed:
Created some more views.

aj-michael committed:
Started work on Application view.

aj-michael committed:
Completed UI for objective functions.

aj-michael committed:
Organized application.conf.

aj-michael committed:
Removed static HTML.

aj-michael commented:
OK, the UI and functionality is done. The only thing left is to merge SQL files to make it compatible with Dan's server.

aj-michael committed:
Create default cluster when creating application.

aj-michael committed:
Removed unused annotations

aj-michael committed:
Removed unused token code.

aj-michael committed:
Trying to make Travis happy.

aj-michael commented:
Ok, it should all be good now. Going to release soon.

aj-michael committed:
Released 0.1.16

aj-michael commented:
This branch is now live on http://thepathfinder.xyz

aj-michael committed:
Dynamic forms and SSL

For some reason, the Google Compute Instance currently running pathfinder-webserver only supports Dynamic Forms.
